File: app/src/main/java/mattframpton/endclothing_mattframpton/di/components/EndClothingApplicationComponent.java
package mattframpton.endclothing_mattframpton.di.components;

import javax.inject.Singleton;

import dagger.Component;
import mattframpton.endclothing_mattframpton.di.modules.EndClothingServiceModule;
import mattframpton.endclothing_mattframpton.di.modules.NetworkModule;
import mattframpton.endclothing_mattframpton.network.EndClothingService;

@Singleton
@Component (modules = {EndClothingServiceModule.class, NetworkModule.class})
public interface EndClothingApplicationComponent {

	EndClothingService getEndClothingService();
}

File: app/src/main/java/mattframpton/endclothing_mattframpton/di/components/EndClothingApplicationComponent.kt
package mattframpton.endclothing_mattframpton.di.components

import dagger.Component
import mattframpton.endclothing_mattframpton.di.modules.EndClothingServiceModule
import mattframpton.endclothing_mattframpton.di.modules.NetworkModule
import mattframpton.endclothing_mattframpton.network.EndClothingService
import javax.inject.Singleton

@Singleton
@Component(modules = [(EndClothingServiceModule::class), (NetworkModule::class)])
interface EndClothingApplicationComponent {

    val endClothingService: EndClothingService
}
